File: keep-alive.js
// Keep-alive script for Render deployment
// Pings the service every 10 minutes to prevent it from sleeping

const https = require('https');

const URL = 'https://briworld.onrender.com/api/health';
const INTERVAL = 10 * 60 * 1000; // 10 minutes

console.log('Starting keep-alive service for BriWorld...');
console.log(`Pinging ${URL} every ${INTERVAL / 1000} seconds`);

function ping() {
  const timestamp = new Date().toISOString();
  
  https.get(URL, (res) => {
    if (res.statusCode === 200) {
      console.log(`[${timestamp}] ✓ Ping successful (HTTP ${res.statusCode})`);
    } else {
      console.log(`[${timestamp}] ✗ Ping failed (HTTP ${res.statusCode})`);
    }
  }).on('error', (err) => {
    console.log(`[${timestamp}] ✗ Ping error: ${err.message}`);
  });
}

// Initial ping
ping();

// Set interval for subsequent pings
setInterval(ping, INTERVAL);

File: backend/internal/keepalive/keepalive.go
package keepalive

import (
	"log"
	"net/http"
	"os"
	"time"
)

func Start() {
	if os.Getenv("ENV") != "production" {
		return
	}

	go func() {
		// Wait 2 minutes for app to fully start
		log.Println("Keep-alive: waiting 2 minutes for app to be ready...")
		time.Sleep(2 * time.Minute)
		log.Println("Keep-alive service started (10 min interval)")

		// First ping immediately
		ping()

		// Then ping every 10 minutes
		ticker := time.NewTicker(10 * time.Minute)
		defer ticker.Stop()

		for range ticker.C {
			ping()
		}
	}()
}

func ping() {
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Get("https://briworld.onrender.com/api/health")
	if err != nil {
		log.Printf("Keep-alive ping failed: %v", err)
		return
	}
	defer resp.Body.Close()
	log.Printf("Keep-alive ping successful: %d", resp.StatusCode)
}
